### PR TITLE
first steps rewrite

### PR DESCRIPTION
**Pull request by rokroskar**
Rewrite first steps including some new UI parts. 

To-Do:

- [x] add note about locale
- [x] consistency between `mydataset` and `mydata`
- [x] code formatting issue 
- [x] add note about adding the ssh key to gitlab

addresses #186 #229

### CODE REVIEW

**Comment by erbou** [comment]
My environment: Mac OS El Capitan 10.11.6.
Starting from scratch with git installed and my ~/.gitconfig configured.

See  my notes after going through the first steps:

### Our first Renku project
1. Prerequesites
Not that if git is a prerequesite that should be mentioned somewhere.

2. Installing Renku

```bash
$ pip install pipenv  # <-- I did not use brew
$ pipenv install --skip-lock renku
...
ValueError: unknown locale: UTF-8 
```

Apparently this is a [_known bug_](https://github.com/pypa/pipenv/issues/187) in pipenv. We should keep that in mind, and put a reference to the pipenv issue if others have the same problem.

### Create a dataset and import data

1. Consistency issues between mydata and mydataset

```bash
$ renku dataset create mydataset
Creating a dataset ... OK
$ ls -l data
total 0
drwxr-xr-x  3 bouillet  staff  102 May 30 17:08 mydataset
$ cat data/mydata/metadata.yml
cat: data/mydata/metadata.yml: No such file or directory
```

The documentation uses alternatively mydata and mydataset - e.g. create mydataset, ls returns mydata, cat mydata.

```bash
renku dataset add mydataset https://raw.githubusercontent.com/
SwissDataScienceCenter/renku/master/README.rst
```

This should be a one liner.

### Updating results based on new input data

```bash
$ renku update
Traceback (most recent call last):
  File "/Users/bouillet/.local/share/virtualenvs/myproject-PeG4IXrS/bin/renku", line 11, in <module>
    sys.exit(cli())
  File "/Users/bouillet/.local/share/virtualenvs/myproject-PeG4IXrS/lib/python3.6/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/Users/bouillet/.local/share/virtualenvs/myproject-PeG4IXrS/lib/python3.6/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/Users/bouillet/.local/share/virtualenvs/myproject-PeG4IXrS/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/bouillet/.local/share/virtualenvs/myproject-PeG4IXrS/lib/python3.6/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/bouillet/.local/share/virtualenvs/myproject-PeG4IXrS/lib/python3.6/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/Users/bouillet/.local/share/virtualenvs/myproject-PeG4IXrS/lib/python3.6/site-packages/click/decorators.py", line 64, in new_func
    return ctx.invoke(f, obj, *args[1:], **kwargs)
  File "/Users/bouillet/.local/share/virtualenvs/myproject-PeG4IXrS/lib/python3.6/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/Users/bouillet/.local/share/virtualenvs/myproject-PeG4IXrS/lib/python3.6/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/Users/bouillet/miniconda3/lib/python3.6/contextlib.py", line 52, in inner
    return func(*args, **kwds)
  File "/Users/bouillet/.local/share/virtualenvs/myproject-PeG4IXrS/lib/python3.6/site-packages/renku/cli/update.py", line 85, in update
    call(['cwl-runner', str(output_file)])
  File "/Users/bouillet/miniconda3/lib/python3.6/subprocess.py", line 267, in call
    with Popen(*popenargs, **kwargs) as p:
  File "/Users/bouillet/miniconda3/lib/python3.6/subprocess.py", line 709, in __init__
    restore_signals, start_new_session)
  File "/Users/bouillet/miniconda3/lib/python3.6/subprocess.py", line 1344, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'cwl-runner': 'cwl-runner'
```

I repeat the update, just in case:

```bash
$ renku update
Error: The repository is dirty. Please use the "git" command to clean it.

On branch master
Untracked files:
  (use "git add <file>..." to include in what will be committed)

	.renku/workflow/bcfd127f8f094a5696ab5b76ed8fa9f5.cwl

nothing added to commit but untracked files present (use "git add" to track)

Once you have added the untracked files, commit them with "git commit".
```

Not sure what happened, it used to work ¯\\\_(ツ)\_/¯

### Making your project live on the Renku platform

I read through, but I have not tried it my self. I have nothing to report yet.

File: docs/user/firststeps.rst [hunk 1]
```diff
@@ -151,7 +170,196 @@ command:
...
+``Settings`` tab and copy the text next to the ``SSH`` box:
+
+.. image:: ../_static/images/project-settings.png
+
+In your terminal, run the following commands in your project directory:
+
+.. code-block:: console
+
+    $ git remote add origin <copied text from SSH box above>
+    $ git push --set-upstream origin master
```
Comment (leafty) [inline docs/user/firststeps.rst]: Without setting up the keypair?

Comment (rokroskar) [reply docs/user/firststeps.rst]: thanks, added to to-do list

**Comment by rokroskar** [comment]
The requested changes have been implemented. If there are no other concerns to address, this should be merged. cc @erbou @leafty 

@jirikuncar I adapted the part with `update` and `status` to reflect the latest changes in `renku-python` but maybe have a quick look if there is anything else that already looks outdated.

File: docs/user/firststeps.rst [hunk 2]
```diff
@@ -32,6 +30,26 @@ First, create a project directory:
     $ mkdir -p ~/renku-projects/myproject
     $ cd ~/renku-projects/myproject
 
+Install ``renku`` and start the shell:
+
+.. code-block:: console
+
+    $ pipenv install --pre renku
```
Comment (jirikuncar) [inline docs/user/firststeps.rst]: Should we recommend `renku` as core dependency or development (`--dev`) dependency for the user projects?

File: docs/user/firststeps.rst [hunk 3]
```diff
@@ -32,6 +30,26 @@ First, create a project directory:
...
+    $ pipenv shell
+
+.. note::
+
+    You may get a ``ValueError: unknown locale: UTF-8`` - in that case, add
```
Comment (jirikuncar) [inline docs/user/firststeps.rst]: Can we link to https://docs.pipenv.org/diagnose/#valueerror-unknown-locale-utf-8 ?

Comment (rokroskar) [reply docs/user/firststeps.rst]: Sure please make the change.

File: docs/user/firststeps.rst (hunk 2 above)
Comment (rokroskar) [reply docs/user/firststeps.rst]: You're right, technically should be a dev dependency but maybe that's an unnecessary complication at this stage?

Comment (jirikuncar) [reply docs/user/firststeps.rst]: Let's hope that people will not question `--dev` options.

File: docs/user/setup.rst [hunk 4]
```diff
@@ -21,17 +21,23 @@ required.
 Prerequisites
 -------------
 
+* python 3.6
+* pip
```
Comment (jirikuncar) [inline docs/user/setup.rst]: (or `pipenv`)

File: docs/spelling_wordlist.txt [hunk 5]
```diff
@@ -9,14 +9,21 @@ CLI
 coopetition
 deployer
 github
+gitlab
+GitLab
+Gitlab
```
Comment (jirikuncar) [inline docs/spelling_wordlist.txt]: we should use one spelling everywhere if possible

Comment (leafty) [reply docs/spelling_wordlist.txt]: Agreed, official one is GitLab.

Comment (rokroskar) [reply docs/spelling_wordlist.txt]: yes, but the docs may include lowercase `gitlab` somewhere as a part of a command/url etc. So the `spelling_wordlist` should include those possibilities (or cause annoying failing builds).

Comment (leafty) [reply docs/spelling_wordlist.txt]: those are fine i guess

**Comment by rokroskar** [comment]
@jirikuncar @leafty can you please approve and merge if the spelling of gitlab isn't a blocker?

**Comment by leafty** [comment]
@rokroskar thought we should remove `Gitlab`. But will approve if it's not important.

**Comment by rokroskar** [comment]
@leafty got it -- done in ec742155f53a3c1bfe3fc42c01511980962cb664

**Pull request merged**